lib/store/store.ts

```diff
@@ -0,0 +1,340 @@
+    this.targetTasks = { isLoading: false };
+    this.targetTimeEntries = { isLoading: false };
+  }
+
+  async authTarget(credentials: T) {
+    const { error, value: adapter } = await this.asyncAction(
+      this.target,
+      async () => {
+        await this.options.target.adapter.init(credentials);
+        return this.options.target.adapter;
+      }
+    );
+
+    if (error || !adapter)
+      await this.options.credentialStorage.reset(this.options.target.name);
+    else {
+      this.setAlert({
+        type: "success",
+        message: "Successfully authenticated!",
+      });
+      this.getTargetProjects();
+      if (!this.selectedTargetTask) this.setTargetTaskNotSelectedError();
+      this.options.credentialStorage.set(
+        this.options.target.name,
+        adapter.credentials
+      );
+    }
+  }
+
+  async authSource(credentials: S) {
+    const { error, value: adapter } = await this.asyncAction(
+      this.source,
+      async () => {
+        await this.options.source.adapter.init(credentials);
+        return this.options.source.adapter;
+      }
+    );
+
+    if (error || !adapter)
+      await this.options.credentialStorage.reset(this.options.source.name);
+    else {
+      this.setAlert({
+        type: "success",
+        message: "Successfully authenticated!",
+      });
+      this.getSourceTimeEntries();
+      this.options.credentialStorage.set(
+        this.options.source.name,
+        adapter.credentials
+      );
+    }
+  }
+
+  async getTargetProjects() {
+    const { error, value: projects } = await this.asyncAction(
+      this.targetProjects,
+      async () => this.target.value?.getProjects()
+    );
+
+    if (
+      !error &&
+      this.selectedTargetProject &&
+      projects?.map(({ id }) => id).includes(this.selectedTargetProject)
+    ) {
+      this.getTargetTasks(this.selectedTargetProject);
+    }
+  }
+
+  getTargetTasks(projectId: Task["projectId"]) {
+    this.asyncAction(this.targetTasks, async () =>
+      this.target.value?.getTasks(projectId)
+    );
+  }
+
+  async getTargetTimeEntries() {
+    if (!this.selectedTargetTask) this.setTargetTaskNotSelectedError();
+    else {
+      const { error } = await this.asyncAction(
+        this.targetTimeEntries,
+        async () => {
+          const entries = await this.target.value?.getTimeEntries(
+            this.correctedDateRange[0] as Date,
+            this.correctedDateRange[1] as Date
+          );
+          return entries?.filter(
+            (entry) => entry.taskId === this.selectedTargetTask
+          );
+        }
+      );
+
+      if (!error) this.selectAllNonSyncedEntries();
+    }
+  }
+
+  async getSourceTimeEntries() {
+    const { error } = await this.asyncAction(
+      this.sourceTimeEntries,
+      async () => {
+        const entries = await this.source.value?.getTimeEntries(
+          this.correctedDateRange[0] as Date,
+          this.correctedDateRange[1] as Date
+        );
```

Comment:
This makes toggl entries from the last day of the month disappear for me.
While Tick expects and filters by plain date values, Toggl filters the entries being in the datetime interval provided. That means we instead of filtering until `2022-01-31 00:00:00` we have to filter until `2022-01-31 23:59:59`. Let's round it up and just add a day to the end date. Inside the TogglAdapter, not here.

Here, just pass `this.dateRange[0], this.dateRange[1]`

Reply:
I ended up using the `startOfDay` and `endOfDay` functions from `date-fns` in the Toggl adapter. This will be resolved in dates like this: `2022-01-30 23:00:00` - `2022-01-31 22:59:59`. So it has the timezone offset, but Toggl handles these dates precisely, so we should get back every entry we wanted.